cmake/FindCUDACompiler.cmake

```diff
@@ -64,14 +64,19 @@ else()
   list(APPEND CMAKE_MODULE_PATH "${ROCM_HOME}/hip/cmake")
   find_package(HIP 1.5.18494 QUIET MODULE)
   if(HIP_FOUND)
-    set(HCC_PATH "${HIP_ROOT_DIR}")
+    if(HIP_VERSION VERSION_LESS "3.1")
+      set(HCC_PATH "${HIP_ROOT_DIR}")
+    else()
+      set(HCC_PATH "${ROCM_HOME}/hcc")
+    endif()
+
     find_package(HIP MODULE)
     message(STATUS "Found HIP compiler: ${HIP_HIPCC_EXECUTABLE}")
     set(CUDA 1)
     set(HIP 1)
     set(CUDA_COMPILER_EXE ${HIP_HIPCC_EXECUTABLE})
     list(APPEND HIP_HCC_FLAGS "-I${HIP_ROOT_DIR}/include -I${ROCM_HOME}/include -Wno-c99-designator -Wno-macro-redefined -Wno-duplicate-decl-specifier -std=c++${CMAKE_CXX_STANDARD}")
-    list(APPEND HIP_HCC_FLAGS "-pedantic -Wall -Wextra -Wno-sign-compare -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-missing-braces -Wno-gnu-anonymous-struct -Wno-nested-anon-types -Wno-gnu-zero-variadic-macro-arguments")
+    list(APPEND HIP_HCC_FLAGS "-pedantic -Wall -Wextra -Wno-sign-compare -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-missing-braces -Wno-gnu-anonymous-struct -Wno-nested-anon-types -Wno-gnu-zero-variadic-macro-arguments -Wno-deprecated-copy")
```

Comment:
`error: unknown warning option '-Wno-deprecated-copy'; did you mean '-Wno-deprecated'? [-Werror,-Wunknown-warning-option]`
if it's a ROCm3.3-specific flag, we should add a version guard